Doveauth gets overloaded by CI

Logs report "Resource temporarily unavailable":
```
Oct 26 16:57:18 c2 dovecot[997575]: auth-worker(998224): Error: conn unix:auth-worker (pid=997595,uid=103): auth-worker<79>: dict(ci-fxgp92@c2.testrun.org): Failed to lookup key shared/userdb/ci-fxgp92@c2.testrun.org: net_connect_unix(/run/doveauth/doveauth.socket) failed: Resource temporarily unavailable (reply took 0.000 secs (0.000 in dict wait, 0.000 in other ioloops, 0.000 in locks))
Oct 26 16:57:18 c2 dovecot[997575]: lmtp(ci-fxgp92@c2.testrun.org)<998466><AJleNWsfHWdCPA8AwVJQ4w:T3>: Error: auth-master: userdb lookup(ci-fxgp92@c2.testrun.org): Auth USER lookup failed
Oct 26 16:57:18 c2 dovecot[997575]: lmtp(998466): Error: lmtp-server: conn unix:pid=1017932,uid=994 [82]: rcpt ci-fxgp92@c2.testrun.org: Failed to lookup user ci-fxgp92@c2.testrun.org: Internal error occurred. Refer to server log for more information.
Oct 26 16:57:18 c2 dovecot[997575]: lmtp(ci-zvrmfu@c2.testrun.org)<1002743><ODOyHmsfHWf3TA8AwVJQ4w:T7:R2>: msgid=<Kg3sCLa6rYtNfuQdHnn-GX1i@localhost>: saved mail to INBOX
Oct 26 16:57:18 c2 dovecot[997575]: imap-login: Login: user=<ci-f3stdf@c2.testrun.org>, method=PLAIN, rip=74.249.21.255, lip=135.181.204.127, mpid=1018270, TLS, session=</vHHIGQlwctK+RX/>
Oct 26 16:57:18 c2 dovecot[997575]: imap-login: Login: user=<ci-93v4ut@c2.testrun.org>, method=PLAIN, rip=172.183.162.43, lip=135.181.204.127, mpid=1018277, TLS, session=<gt/LIGQlg+ist6Ir>
Oct 26 16:57:18 c2 dovecot[997575]: lmtp(998486): Connect from local
```

Dovecot has documentation for this error: https://doc.dovecot.org/2.3/admin_manual/errors/socket_unavailable/

`doveconf` reports that client_limit for `imap` is 1, so default_client_limit is not used, there is no problem with the client_limit.

`imap` has `process_limit=50000`.
`sysctl net.core.somaxconn` reports 4096.

CI fails like this:
```
 command_result = ('NO', [b'[UNAVAILABLE] Temporary authentication failure. [c2:2024-10-26 16:54:13]'])
exception = <class 'imap_tools.errors.MailboxLoginError'>, expected = 'OK'

    def check_command_status(command_result: tuple, exception: type, expected='OK'):
        """
        Check that IMAP command responses status equals <expected> status
        If not, raise specified <exception>
        :param command_result: imap command result: tuple(typ, data)
        :param exception: exception subclass of UnexpectedCommandStatusError, that raises
        :param expected: expected command status
        """
        typ, data = command_result[0], command_result[1]
        if typ != expected:
>           raise exception(command_result=command_result, expected=expected)
E           imap_tools.errors.MailboxLoginError: Response status "OK" expected, but "NO" received. Data: [b'[UNAVAILABLE] Temporary authentication failure. [c2:2024-10-26 16:54:13]']
```

See also https://github.com/deltachat/deltachat-core-rust/issues/6065

Looks like [`request_queue_size`](https://docs.python.org/3.12/library/socketserver.html#socketserver.BaseServer.request_queue_size) was lost in commit bf0f6e230352b9a05fa5b2e4ce59e511731f412e (https://github.com/deltachat/chatmail/pull/377)

It was added in https://github.com/deltachat/chatmail/pull/82, but then doveauth got faster thanks to not using SQLite, then request_queue_size was lost and this was not noticed because the queue does not overflow on a fast server. This c2.testrun.org is the cheapest VPS possible so it cannot handle requests without a queue.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,9 @@
 
 ## untagged
 
+- increase `request_queue_size` for UNIX sockets to 1000.
+  ([#437](https://github.com/deltachat/chatmail/pull/437))
+
 - query autoritative nameserver to bypass DNS cache
   ([#424](https://github.com/deltachat/chatmail/pull/424))
 

--- a/chatmaild/src/chatmaild/dictproxy.py
+++ b/chatmaild/src/chatmaild/dictproxy.py
@@ -87,8 +87,12 @@ class DictProxy:
         except FileNotFoundError:
             pass
 
-        with ThreadingUnixStreamServer(socket, Handler) as server:
+        with CustomThreadingUnixStreamServer(socket, Handler) as server:
             try:
                 server.serve_forever()
             except KeyboardInterrupt:
                 pass
+
+
+class CustomThreadingUnixStreamServer(ThreadingUnixStreamServer):
+    request_queue_size = 1000

